File: tracker/src/test/java/com/cyrillrx/tracker/DummyScheduledTracker.java
package com.cyrillrx.tracker;

import com.cyrillrx.tracker.event.TrackEvent;

import java.util.ArrayList;
import java.util.Collection;
import java.util.List;
import java.util.concurrent.TimeUnit;

/**
 * A basic tracker the adds the tracked event to a list.
 *
 * @author Cyril Leroux
 *         Created on 22/04/2016.
 */
class DummyScheduledTracker extends ScheduledTracker {

    public final List<TrackEvent> trackedEvents = new ArrayList<>();

    public DummyScheduledTracker(long timeDuration, TimeUnit timeUnit, int workerCount) {
        super(timeDuration, timeUnit, workerCount);
    }

    public DummyScheduledTracker(long timeDuration, TimeUnit timeUnit) {
        super(timeDuration, timeUnit);
    }

    @Override
    protected void consumeEvents(Collection<TrackEvent> events) {
        trackedEvents.addAll(events);
    }

    boolean isEmpty() { return trackedEvents.isEmpty(); }

    int getEventCount() { return trackedEvents.size(); }
}

File: tracker/lib/src/androidUnitTest/kotlin/com/cyrillrx/tracker/DummyScheduledTracker.kt
package com.cyrillrx.tracker

import com.cyrillrx.tracker.event.TrackEvent
import java.util.concurrent.TimeUnit

/**
 * A basic tracker the adds the tracked event to a list.
 *
 * @author Cyril Leroux
 * Created on 22/04/2016.
 */
internal class DummyScheduledTracker(
    timeDuration: Long,
    timeUnit: TimeUnit,
    workerCount: Int = DEFAULT_WORKER_COUNT
) : ScheduledTracker(timeDuration, timeUnit, workerCount) {
    val trackedEvents: MutableList<TrackEvent> = ArrayList()

    protected override fun consumeEvents(events: Collection<TrackEvent>) {
        trackedEvents.addAll(events)
    }

    fun isEmpty(): Boolean = trackedEvents.isEmpty()

    fun getEventCount(): Int = trackedEvents.size
}
